packages/astro/src/assets/vite-plugin-assets.ts

```diff
@@ -107,13 +107,13 @@ export default function assets({
 						}
 
 						const url = new URL(req.url, 'file:');
-						const filePath = url.searchParams.get('href');
+						const filePath = url.searchParams.get('href')?.slice('/@fs'.length);
 
 						if (!filePath) {
 							return next();
 						}
 
-						const filePathURL = new URL('.' + filePath, settings.config.root);
+						const filePathURL = new URL('.' + filePath, 'file:');
```

Comment:
Won't this make it so that it's always relative to your filesystem root?

Reply:
Maybe it's becoming relative to the `pwd`?

Reply:
> Won't this make it so that it's always relative to your filesystem root?

It's always relative to root, but this is not a problem as far as I could tell